src/shared/gitlab-client.ts

```diff
@@ -149,4 +159,77 @@ export class GitlabClient {
     };
   }
 
+  public async getUserByEmail(email: string) {
+    if (!validateEmail(email)) {
+      throw Boom.badRequest(`Invalid email '${email}'`);
+    }
+    const search = {
+      search: email,
+    };
+    const users = await this.fetchJson<User[]>(`users?${qs.stringify(search)}`, true);
+    if (!users || !users.length) {
+      throw Boom.badRequest(`Can\'t find user '${email}'`);
+    }
+    if (users.length > 1) {
+      // This shoud never happen
+      throw Boom.badRequest(`Found multiple users with email '${email}'`);
+    }
+    return users[0];
+  }
+
+  public createUser(
```

Comment:
OK.

I think there are benefits for both approaches.

I would like that code that is *only* used by authentication-module (or the related hapi-plugin) would be inside the `authentication` folder.

Do you think that the benefits of the thick-Gitlab-client approach compared to the thin-Gitlab-client approach is so clear that it makes sense to switch away from previous practice?